jersey/connector/src/test/java/io/helidon/jersey/connector/MultipartTest.java

```diff
@@ -0,0 +1,105 @@
+import org.glassfish.jersey.message.internal.ReaderWriter;
+import org.junit.jupiter.api.Assertions;
+
+import org.junit.jupiter.api.Test;
+
+import javax.ws.rs.Consumes;
+import javax.ws.rs.POST;
+import javax.ws.rs.Path;
+import javax.ws.rs.client.ClientBuilder;
+import javax.ws.rs.client.Entity;
+import javax.ws.rs.client.WebTarget;
+import javax.ws.rs.core.Application;
+import javax.ws.rs.core.Context;
+import javax.ws.rs.core.HttpHeaders;
+import javax.ws.rs.core.MediaType;
+import javax.ws.rs.core.Response;
+
+import java.io.IOException;
+import java.util.HashSet;
+import java.util.Set;
+
+@HelidonTest
+@DisableDiscovery
+@AddExtension(ServerCdiExtension.class)
+@AddExtension(JaxRsCdiExtension.class)
+@AddExtension(CdiComponentProvider.class)
+
+@AddBean(MultipartTest.MultipartApplication.class)
+public class MultipartTest {
+    private static final String ENTITY = "hello";
+    public static class MultipartApplication extends Application {
+        @Override
+        public Set<Class<?>> getClasses() {
+            Set<Class<?>> set = new HashSet<>();
+            set.add(MultiPartFeature.class);
+            set.add(MultipartResource.class);
+            return set;
+        }
+    }
+
+    @Path("/")
+    public static class MultipartResource {
+        @POST
+        @Path("upload")
+        @Consumes(MediaType.MULTIPART_FORM_DATA)
+        public String upload(@Context HttpHeaders headers, MultiPart multiPart) throws IOException {
+//            headers.getRequestHeaders().forEach((k,v) -> System.out.println(k + ":" + v));
+            return ReaderWriter.readFromAsString(
+                    ((BodyPartEntity) multiPart.getBodyParts().get(0).getEntity()).getInputStream(),
+                    MediaType.TEXT_PLAIN_TYPE);
+        }
+    }
+
+    @Test
+    void testMultipart(WebTarget webTarget) {
+        // For each entity type
+        for (HelidonEntity.HelidonEntityType entityType : HelidonEntity.HelidonEntityType.values()) {
+            // make 10 consecutive requests
+            for (int i = 0; i != 10; i++) {
+//                System.out.append(entityType.name()).println(i);
+                MultiPart multipart = new MultiPart().bodyPart(new BodyPart().entity(ENTITY));
+                multipart.setMediaType(MediaType.MULTIPART_FORM_DATA_TYPE);
+                try (Response r = ClientBuilder.newBuilder()
+                        .property(HelidonConnector.INTERNAL_ENTITY_TYPE, entityType.name())
+                        .build().target(webTarget.getUri())
+                        .path("upload")
+                        .register(MultiPartFeature.class)
+                        .request()
+                        .post(Entity.entity(multipart, multipart.getMediaType()))) {
+                    Assertions.assertEquals(Response.Status.OK.getStatusCode(), r.getStatus());
+                    Assertions.assertEquals(ENTITY, r.readEntity(String.class));
+                }
+            }
+        }
+    }
```

Comment:
Please use `@RepeatedTest` or `@ParameterizedTest` with `@EnumSource` 
```suggestion
    @ParameterizedTest
    @EnumSource(value = HelidonEntity.HelidonEntityType.class)
    void testMultipart(HelidonEntity.HelidonEntityType entityType, WebTarget webTarget) {
        // For each entity type
        // make 10 consecutive requests
        for (int i = 0; i != 10; i++) {
            MultiPart multipart = new MultiPart().bodyPart(new BodyPart().entity(ENTITY));
            multipart.setMediaType(MediaType.MULTIPART_FORM_DATA_TYPE);
            try (Response r = ClientBuilder.newBuilder()
                    .property(HelidonConnector.INTERNAL_ENTITY_TYPE, entityType.name())
                    .build().target(webTarget.getUri())
                    .path("upload")
                    .register(MultiPartFeature.class)
                    .request()
                    .post(Entity.entity(multipart, multipart.getMediaType()))) {
                Assertions.assertEquals(Response.Status.OK.getStatusCode(), r.getStatus());
                Assertions.assertEquals(ENTITY, r.readEntity(String.class));
            }
        }
    }
```